src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -176,6 +184,29 @@ AssemblyMetadata GetMetadataWorker(string fullPath)
 
             return metadata;
         }
+
+        void ClearReleasedMetadata_NoLock()
+        {
+            Contract.ThrowIfFalse(Monitor.IsEntered(_metadataCacheLock));
+
+            var count = _metadataCache.Count;
+
+            // Cleanup when we hit powers of two.  This way we don't have to walk too often.  But only when we've really
+            // grown. the dictionary by a substantial amount since the last time we cleaned up.
+            if (count == (1 << SegmentedArraySortUtils.Log2((uint)count)))
```

Comment:
this basically preserves the prior logic we had where we'd cleanup when we hit powers-of-two capacity.  i don't love this approach.  But it worked before, so i'm ok rolling back to that.